### PR TITLE
Spring clean + move to Github Actions

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset refreshes things a little.

* drops support for end-of-life Python and Django versions
   * refactors to stop using `django.utils.deprecation.MiddlewareMixin`, which is unnecessary with LTS and latest (4.0) Django
* moves CI to Github Actions

### Follow-up actions:
- [ ] A new release of this new `0.4.0` version should be done on PyPi

### To test 
- Check Actions/CI
- Try installing it in a local build of something that uses it

### CODE REVIEW

File: AUTHORS.rst [hunk 1]
```diff
@@ -10,4 +10,6 @@ Development Lead
 Contributors
 ------------
 
-None yet. Why not be the first?
+* Steve Jalim <stevejalim@mozilla.com>
```
Comment (pmac) [inline AUTHORS.rst:13]: You sure you want your email this public?

**Review by pmac — APPROVED** [review]
Looks nice! Did you also run `black` on it?

File: AUTHORS.rst (hunk 1 above)
Comment (stevejalim) [reply AUTHORS.rst]: Not too fussed, but swapped for a Firefox Relay address - will be interesting to see what comes in

**Comment by stevejalim** [comment]
@pmac good call re auto-formatting -- just `black`ed it now in [c46a716](https://github.com/mozmeao/django-allow-cidr/pull/17/commits/c46a71691b9c5cf3cdc1c5475f3b88d71a09516f) - minor changes

**Pull request merged**